core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -0,0 +1,142 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.hadoop;
+
+import com.google.common.base.Preconditions;
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Map;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.iceberg.BaseMetastoreCatalog;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.TableMetadata;
+import org.apache.iceberg.TableOperations;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.exceptions.AlreadyExistsException;
+import org.apache.iceberg.exceptions.RuntimeIOException;
+
+
+public class HadoopCatalog extends BaseMetastoreCatalog implements Closeable {
+  private static final String ICEBERG_HADOOP_WAREHOUSE_BASE = "iceberg/warehouse";
+  private final Configuration conf;
+  private String hdfsRoot;
+
+  public HadoopCatalog(Configuration conf) {
+    this.conf = conf;
+    hdfsRoot = conf.get("fs.defaultFS");
+    Path warehousePath = new Path(hdfsRoot + ICEBERG_HADOOP_WAREHOUSE_BASE);
+    try {
+      FileSystem fs = Util.getFs(warehousePath, conf);
+      if (!fs.isDirectory(warehousePath)) {
+        if (!fs.mkdirs(warehousePath)) {
+          throw new IOException("failed to create warehouse for hadoop catalog");
+        }
+      }
+      this.hdfsRoot = hdfsRoot + "/" + ICEBERG_HADOOP_WAREHOUSE_BASE;
+    } catch (IOException e) {
+      throw new RuntimeIOException("failed to create directory for warehouse", e);
+    }
+  }
+
+  @Override
+  public org.apache.iceberg.Table createTable(
+      TableIdentifier identifier, Schema schema, PartitionSpec spec, String location, Map<String, String> properties) {
+    Preconditions.checkArgument(identifier.namespace().levels().length == 1,
+            "Missing database in table identifier: %s", identifier);
+    Path tablePath = new Path(defaultWarehouseLocation(identifier));
+    try {
+      FileSystem fs = Util.getFs(tablePath, conf);
+      if (!fs.isDirectory(tablePath)) {
+        fs.mkdirs(tablePath);
+      } else {
+        throw new AlreadyExistsException("the table already exists: " + identifier);
+      }
+    } catch (IOException e) {
+      throw new RuntimeIOException("failed to create directory", e);
+    }
+    return super.createTable(identifier, schema, spec, null, properties);
+  }
+
+  public org.apache.iceberg.Table createTable(
+          TableIdentifier identifier, Schema schema, PartitionSpec spec, Map<String, String> properties) {
+    Preconditions.checkArgument(identifier.namespace().levels().length == 1,
+            "Missing database in table identifier: %s", identifier);
+    return createTable(identifier, schema, spec, null, properties);
```

Comment:
can we have `properties` as the 4th parameter, as passing `null` in the middle looks bit odd.
this might need fix to `BaseMetastoreCatalog.createTable` as well.

Reply:
That needs to change the Catalog interface as well, and thus need change for HiveCatalog. So I would leave it to avoid too many changes not related.

Reply:
sure, make sense. 
we can create a different issue to fix this, in a separate PR.
@rdblue is it ok with you ?

Reply:
Im sorry for this.

When I think more about it, it should be ok. As parent class signature would not change based on the child class behavior. Also parent method is expecting `location` and `properties` to be null.